toolsrc/src/vcpkg/base/json.cpp

```diff
@@ -1292,9 +1323,20 @@ namespace vcpkg::Json
         }
 
         auto extra_fields = invalid_json_fields(obj, valid_fields);
-        if (!extra_fields.empty())
+        for (auto&& f : extra_fields)
         {
-            add_extra_fields_error(type_name.to_string(), std::move(extra_fields));
+            size_t best_index = 0;
+            auto best_value = Strings::byte_edit_distance(f, valid_fields[0]);
+            for (size_t i = 1; i < valid_fields.size(); ++i)
+            {
+                auto v = Strings::byte_edit_distance(f, valid_fields[i]);
+                if (v < best_value)
+                {
+                    best_value = v;
+                    best_index = i;
+                }
+            }
```

Comment:
```suggestion
            const auto last = valid_fields.end();
            auto best_it = valid_fields.begin();
            auto best_value = Strings::byte_edit_distance(f, *best_it);
            for (auto it = best_it + 1; it != last; ++it)
            {
                auto v = Strings::byte_edit_distance(f, *it);
                if (v < best_value)
                {
                    best_value = v;
                    best_it = it;
                }
            }
```

I'd prefer you used iterators here.